### PR TITLE
report_creation_index_boundaries_check

### PR DESCRIPTION
**Pull request by duskoo**
Several tests for index boundaries with resulting tcns vector equality assertion.

### CODE REVIEW

File: tests/basic_functionality.rs [hunk 1]
```diff
@@ -257,3 +259,71 @@ fn report_with_j2_max_and_j1_max_minus_1_generates_2_tcns() {
     let recomputed_tcns = report.temporary_contact_numbers().collect::<Vec<_>>();
     assert_eq!(recomputed_tcns.len(), 2);
 }
+
+fn tcn_generation_test_helper(rak: ReportAuthorizationKey, tcns: Arc<Vec<TemporaryContactNumber>>, j_1: u16, j_2: u16){
```
Comment (i-schuetz) [inline tests/basic_functionality.rs:263]: Just curious, can't this be a reference instead of `Arc`? Also, why `Arc` instead of `Rc` (not thread safe)?

Comment (duskoo) [reply tests/basic_functionality.rs]: Arc was an overkill

**Comment by i-schuetz** [comment]
Can you open this PR in the original repo? Probably it makes sense to remove the print statements.

Sorry for the late review.

**Comment by duskoo** [comment]
> Can you open this PR in the original repo? Probably it makes sense to remove the print statements.
> 
> Sorry for the late review.

Opened PR agains the main repo: https://github.com/TCNCoalition/TCN/pull/82

**Pull request closed without merging**